src/doc/specifying-dependencies.md

```diff
@@ -164,12 +164,59 @@ hello_utils = { path = "hello_utils", version = "0.1.0" }
 
 # Overriding Dependencies
 
-Sometimes you may want to override one of Cargo’s dependencies. For example,
-let’s say you’re working on a project,
+Sometimes you may want to override one of Cargo’s dependencies. For example
+let's say you're working on a project using the
+[`uuid`](https://crates.io/crates/uuid) crate which depends on
+[`rand`](https://crates.io/crates/rand). You've discovered there's a bug in
+`rand`, however, and it's already fixed upstream but hasn't been published yet.
+You'd like to test out the fix, so let's first take a look at what your
+`Cargo.toml` will look like:
+
+```toml
+[package]
+name = "my-awesome-crate"
+version = "0.2.0"
+authors = ["The Rust Project Developers"]
+
+[dependencies]
+uuid = "0.2"
+```
+
+To override the `rand` dependency of `uuid`, we'll leverage the [`[replace]`
+section][replace-section] of `Cargo.toml` by appending this to the end:
+
+[replace-section]: manifest.html#the-replace-section
+
+```toml
+[replace]
+"rand:0.3.14" = { git = 'https://github.com/rust-lang-nursery/rand' }
+```
+
+This indicates that the version of `rand` we're currently using, 0.3.14, will be
+replaced with the master branch of `rand` on GitHub. Next time when you execute
+`cargo build` Cargo will take care of checking out this repository and hooking
+the `uuid` crate up to the new version. Note that the `[replace]` section
+requires that crates are replaced with other copies that have both the same name
+and version (but they can come from different sources).
```

Comment:
I found it difficult here to imagine how to use this feature effectively. What I would expect is that I can use `replace` to substitute my dependency for a yet-unpublished fix upstream, but this isn't strictly true. There's a good possibility that upstream has already bumped their version number (there are e.g. release tools in the cargo ecosystem that bump the version immediately after publishing a release), making it replaceable.

The workflow for using `replace` seems quite constrained, and works best when you control the replacement code yourself. It might make sense to walk through a real case.